Maintenance Request Oxmaint (Independent Publisher)

JRS-Connector opened the pull request:
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [ ] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [ ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

JRS-Connector committed:
Adding Maintenance Request Oxmaint

:wq

vmanoharas commented:
Hello @JRS-Connector, 

Kindly agree the CLA and follow below documents and submit your connector. 

Settings.json and icon.png files are not required to submit. 

https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission-ip
https://github.com/microsoft/PowerPlatformConnectors/tree/dev/templates/Independent%20Publisher

JRS-Connector commented:
@microsoft-github-policy-service agree [company="Oxmaint Inc."]

JRS-Connector commented:
@microsoft-github-policy-service agree
@microsoft-github-policy-service agree company="Oxmaint Inc."

vmanoharas commented:
> Hello @JRS-Connector,
> 
> Kindly agree the CLA and follow below documents and submit your connector.
> 
> Settings.json and icon.png files are not required to submit.
> 
> https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission-ip https://github.com/microsoft/PowerPlatformConnectors/tree/dev/templates/Independent%20Publisher

Hi @JRS-Connector, 

Kindly address above comments and re submit this PR.

JRS-Connector committed:
Update Readme.md

JRS-Connector commented:
Hello @vmanoharas,
The changes have been addressed in the same PR. Do I have to create a new PR?

vmanoharas commented:
Hello @JRS-Connector, 

Thank you for your reply, I can still see the settings.json and icon.png files in the artifacts, please see the screenshot below. 

![image](https://github.com/user-attachments/assets/2b00a716-1039-4d84-8a96-61b39cbd9a13)

JRS-Connector committed:
Delete independent-publisher-connectors/MaintenanceRequestOxmaint/settings.json

JRS-Connector committed:
Delete independent-publisher-connectors/MaintenanceRequestOxmaint/icon.png

JRS-Connector commented:
Hello @vmanoharas, Sorry about that! I have made the changes now. Could you please have a look?

vmanoharas commented on independent-publisher-connectors/MaintenanceRequestOxmaint/apiProperties.json:
```diff
@@ -0,0 +1,8 @@
+{
+  "properties": {
+    "connectionParameters": {},
+    "iconBrandColor": "#ffffff",
```
IconBrandColor should not be Use a non-transparent, nonwhite color (#ffffff) background, and not-default color (#007ee5) that matches your specified icon background color. However for Independent Publisher the color should be mentioned in the below url 

https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/apiProperties.json

vmanoharas commented on independent-publisher-connectors/MaintenanceRequestOxmaint/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,147 @@
+{
+  "swagger": "2.0",
```
Please refer the below link and add contact and x-ms-connector-metadata property. 

https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/apiDefinition.swagger.json

vmanoharas commented on independent-publisher-connectors/MaintenanceRequestOxmaint/Readme.md:
```diff
@@ -0,0 +1,47 @@
+# Request Maintenance Oxmaint Connector
```
Please include all the required sections into the readme.md file

JRS-Connector replied:
Hi @vmanoharas, I have included all the required sections viz. Title, Publisher, Prerequisites, supported operations, obtaining credentials, Known Issues and Limitations. Please let me know if you're looking for a specific section I might have missed.

vmanoharas reviewed (commented):
Hello @jrs-connector

Kindly resolve review comments.

JRS-Connector committed:
Update iconBrandColor in apiProperties.json

JRS-Connector committed:
Updated contact information in apiDefinition.swagger.json

vmanoharas commented on independent-publisher-connectors/MaintenanceRequestOxmaint/apiProperties.json:
```diff
@@ -0,0 +1,8 @@
+{
+  "properties": {
+    "connectionParameters": {},
+    "iconBrandColor": "#f7f7f7",
```
Kindly refer below link and change the iconBrandColor, which is standard color for Independent Publisher.

vmanoharas commented on independent-publisher-connectors/MaintenanceRequestOxmaint/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,152 @@
+  "tags": [],
+  "x-ms-openai-manifest": {
+    "humanName": ""
+  }
```
Kindly refer the below link and add "x-ms-connector-metadata" property

https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/apiDefinition.swagger.json#L25

vmanoharas reviewed (commented):
Hello @JRS-Connector, 

Kindly resolve review comments.

JRS-Connector committed:
Update apiDefinition.swagger.json

"x-ms-connector-metadata" details have been updated

JRS-Connector commented:
x-ms-connector-metadata has been updated.

And please let me know about the issue with iconColor

vmanoharas commented:
Apologies for missing the link, please refer the link below and update the iconBrandColor. 

https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/apiProperties.json

JRS-Connector committed:
Update apiProperties.json

I have updated the IconBrandColor and also i have updated the other issue also. Kindly Look over it

JRS-Connector commented:
@vmanoharas The changes have been done.

vmanoharas commented:
> @vmanoharas The changes have been done.

Hi @JRS-Connector, 

The IconBrandColor is not updated as per the documentation. The color is standard for independent connector as "da3b01", please update. 

![image](https://github.com/user-attachments/assets/4feee326-feaa-47c3-9187-487bd73737c6)

JRS-Connector committed:
Update apiProperties.json

Update has been done could you please check it

JRS-Connector commented:
I have updated the IconBrandColor as per your suggestion. Could you please review it.

vmanoharas reviewed (approved):
Dear Partner

I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks.
Please let us know if you have any questions.
Thank you very much for working with us.

vmanoharas commented:
[[certify-connector]]

JRS-Connector commented:
Could you please let me Know what is the further process so that I do that.

vmanoharas commented:
Hello @JRS-Connector, 

You don't have to take any action further, we will process your connector and deploy to all the regions in next 3 to 4 weeks.

JRS-Connector commented:
Can you please provide update over this connector as in the mail I am unable to open it and get over the mail data. If there is anything in the email that would need action from my side you can drop the mail at: "sachin@oxmaint.com"

vmanoharas commented:
Hello Partner,
I hope you are doing well.
Congratulations, the deployment of your connector and documentation page was already completed, We have a short survey - https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program.
Thank you very much for working with us.
